### PR TITLE
Fix sigterm triggered cancellation

### PR DESCRIPTION
**Pull request by aneojgurhem**
fix https://github.com/aneoconsulting/ArmoniK/issues/479

### CODE REVIEW

File: Common/src/Utils/GraceDelayCancellationTokenSource.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+{
+  private readonly TimeSpan                      t1_;
+  private readonly TimeSpan                      t2_;
+  private readonly TimeSpan                      t3_;
+  private readonly TimeSpan                      t4_;
+  private readonly TimeSpan                      t5_;
+  private readonly CancellationTokenRegistration reg_;
+
+  public GraceDelayCancellationTokenSource(CancellationTokenSource source,
+                                           TimeSpan                t1,
```
Comment (lemaitre-aneo) [inline Common/src/Utils/GraceDelayCancellationTokenSource.cs:40]: 5 timespan?

Comment (aneojgurhem) [reply Common/src/Utils/GraceDelayCancellationTokenSource.cs:40]: why not ? We made a slightly more general class

Comment (lemaitre-aneo) [reply Common/src/Utils/GraceDelayCancellationTokenSource.cs:40]: I would say, if you want more delays, you create more instances of `GraceDelayCancellationTokenSource`.

Comment (aneojgurhem) [reply Common/src/Utils/GraceDelayCancellationTokenSource.cs:40]: We need one instance with two delays

**Review by lemaitre-aneo — APPROVED** [review]

File: Common/src/Pollster/AgentHandler.cs [hunk 2]
```diff
@@ -70,6 +71,7 @@ public class AgentHandler : IAgentHandler, IAsyncDisposable
                       IObjectStorageFactory objectStorageFactory,
                       ILogger<AgentHandler> logger)
   {
+    computePlanOptions_   = computePlanOptions;
```
Comment (lemaitre-aneo) [inline Common/src/Pollster/AgentHandler.cs:74]: `ComputePlaneOptions`

File: Compute/PollingAgent/src/Program.cs [hunk 3]
```diff
@@ -72,7 +72,8 @@ public static int Main(string[] args)
 
     try
     {
-      builder.Host.UseSerilog(logger.GetSerilogConf());
+      builder.Host.UseSerilog(logger.GetSerilogConf())
+             .ConfigureHostOptions(options => options.ShutdownTimeout = TimeSpan.FromDays(1));
```
[the remark this answers is not in the document]
Comment (aneojgurhem) [reply Compute/PollingAgent/src/Program.cs:76]: It conflicts with our gracedelay implementation.

[the remark this answers is not in the document]
Comment (aneojgurhem) [reply Compute/PollingAgent/src/Program.cs:76]: I'm putting a parameter for the tilmeout

**Review by lemaitre-aneo — APPROVED** [review]

**Pull request merged**